discovery/manager.go

```diff
@@ -418,7 +443,11 @@ func (m *Manager) registerProviders(cfg sd_config.ServiceDiscoveryConfig, setNam
 	}
 	for _, c := range cfg.AzureSDConfigs {
 		add(c, func() (Discoverer, error) {
-			return azure.NewDiscovery(c, log.With(m.logger, "discovery", "azure")), nil
+			d, err := GetDiscoverer("azure")
```

Comment:
Can we adjust `GetDiscoverer` in a way that would not require this anonymous function? :thinking: